asdf/yamlutil.py

```diff
@@ -405,6 +405,15 @@ def dump_tree(tree, fd, ctx, tree_finalizer=None, _serialization_context=None):
     yaml_version = tuple(
         int(x) for x in ctx.version_map['YAML_VERSION'].split('.'))
 
+    # add yaml %TAG definitions from extensions
+    if _serialization_context:
+        for ext in _serialization_context._extensions_used:
+            ext_tags = getattr(ext.delegate, "yaml_tag_handle", None)
```

Comment:
With the changes I recommended above, we should be able to simplify to this:

```python
for ext in _serialization_context._extensions_used:
    for key, val in ext.yaml_tag_handles.items():
        if key not in tags:
            tags[key] = val
```

Reply:
Up above it looks like it's possible for `tags` to still be None when we get here.  Maybe we should initialize it with an empty dict so we don't risk a problem?

Reply:
I wasn't sure in what context this part would not get executed
https://github.com/asdf-format/asdf/blob/62fe49a9032e0883c0b0f5ebe0f51c7b4f6e72de/asdf/yamlutil.py#L388-L398

but you are right, I will init with empty dict and change accordingly

Reply:
Hmm... you're right, and the [standard](https://asdf-standard.readthedocs.io/en/1.5.0/tree.html#tags) says:

> ... the tag of the root element of the tree must always be tag:stsci.edu:asdf/core/asdf-1.1.0

So if tree_type is `None`, that should probably be an error.  I'll file a separate issue to clean that up.